Eclipse-Laboratories-Inc/eclipse-bridge-ux
# Kyle/implement mint

# Nucleus Integration
Implements Nucleus minting in Eclipse.

[Nucleus](https://www.nucleusearn.io/) provides the ability to mint and bridge tETH to the Eclipse network using one of our accepted assets. This PR provides that functionality directly into the Eclipse UI through a new page on the Eclipse app. 
![image](https://github.com/user-attachments/assets/19e67289-036a-4ddf-ae20-782893a5f3d9)


### Notes 
- New route called `/mint` was created. This should be changed to whatever it needs to be. 
- On the Figma design there is a nav bar on the left that doesn't exist on prod. I was told this is currently being developed by the Eclipse team. The new `/mint` route will need to be added to that nav bar when it is complete.
- All code changes are in the `mint` directory under `app/mint`.
- Some new assets have been added to the public folder. 
- You will notice some changes in the diff outside of the mint directory, but these are just new lines and empty spaces caused by prettier. 
- The `Mint.tsx` component is one big component for simplicity. It could easily be broken down into smaller pieces. 
- Implemented the mint page based on the figma provided below
- Implemented the transaction modal based on the production transaction modal
- I used a new "mint" version of the transaction modal since I wanted to implement a simpler version that is not tied to any context or backend api.

### Testing
Note that testing only works on mainnet for now. Our contracts are deployed on Tenderly but I was having trouble getting Dynamic to connect to Tenderly. So testing will have to be done on mainnet.

1. Ensure your wallet has a balance in one of our accepted assets: ezETH, apxETH, rswETH, weETH, or WETH
2. Navigate to http://localhost:3000/mint
3. Select the token asset you want to deposit
4. Enter a deposit amount into the input box 
5. Click "Mint"
6. You will be prompted to approve the contract to use your tokens
7. If your wallet prompts you to sign the deposit message, this means that simulate passed so you can feel pretty confident that the transaction will work. You can reject the transaction at this point if you don't want to pay gas fees.
8. If you want to follow through and pay gas fees, then you can do so and verify that the transaction modal shows the correct information. 
  
### Figma
- https://www.figma.com/design/9h3HDcACU0JleBUKmduR7d/Eclipse-%3C%3E-Nucleus?node-id=1-274&node-type=frame&t=52sKp3H2msnV4ocf-11

### More Screenshots
Amount over balance
![image](https://github.com/user-attachments/assets/ba95c828-c900-47c5-a90b-560df45cb3d4)
Deposit pending
![image](https://github.com/user-attachments/assets/31bf12f4-f4c4-4c3b-ab6e-cf6db86093b9)
Deposit complete
![image](https://github.com/user-attachments/assets/958d227f-c663-4456-91ee-f0c684b7dc17)
